### ISSUE TITLE
Unexpected behaviour for size in UpdateByQuery

### ISSUE THREAD

**Issue by ioanatia** [elastic/elasticsearch#63005]
**Elasticsearch version**

Version: 7.9.2, Build: default/tar/d34da0ea4a966c4e49417f2da2f244e3e97b4e6e/2020-09-23T00:45:33.626720Z, JVM: 15

**JVM version**

javac 1.8.0_172

**Description of the problem including expected versus actual behavior**:

In 7.x the `size` parameter was deprecated in favour of `max_docs`.
When `size` is passed as a query parameter, Elasticsearch gives a deprecation message, but it still takes `size` into account.

However, when `size` is passed in the body of the request, the behaviour is completely different.
No deprecation message is trigerred.
Furthermore Elasticsearch just updates all the documents that match the query.
A closer look also shows that in this case `size` acts more like `scroll_size`, because it seems to dictate the batch size.

When `size` is passed in the query, it works as expected:
```
POST my-index/_update_by_query?size=10
{
  "query": {
    "term": {
      "year$string": 2007
    }
  },
  "script": {
    "source": "ctx._source['year$string'] = 3000;"
  }
}

#! Deprecation: Deprecated field [size] used, expected [max_docs] instead
{
  "took" : 23,
  "timed_out" : false,
  "total" : 10,
  "updated" : 10,
  "deleted" : 0,
  "batches" : 1,
  "version_conflicts" : 0,
  "noops" : 0,
  "retries" : {
    "bulk" : 0,
    "search" : 0
  },
  "throttled_millis" : 0,
  "requests_per_second" : -1.0,
  "throttled_until_millis" : 0,
  "failures" : [ ]
}
```

When `size` is passed in the body, all docs are updated and `size` seems to act more like the batch size:

```
POST my-index/_update_by_query
{
  "query": {
    "term": {
      "year$string": 2007
    }
  },
  "script": {
    "source": "ctx._source['year$string'] = 3000;"
  },
  "size": 10
}

{
  "took" : 488,
  "timed_out" : false,
  "total" : 354,
  "updated" : 354,
  "deleted" : 0,
  "batches" : 36,
  "version_conflicts" : 0,
  "noops" : 0,
  "retries" : {
    "bulk" : 0,
    "search" : 0
  },
  "throttled_millis" : 0,
  "requests_per_second" : -1.0,
  "throttled_until_millis" : 0,
  "failures" : [ ]
}

```

Is this the expected behaviour?
I'd expect to at least see a deprecation message.
It would also be interesting to see whether in 8.x you can still pass `size` in the body, since its usage should no longer be supported.

**Comment by henningandersen** [comment]
Yes, this is indeed a bit confusing. The behavior you see is expected but not intuitive.

`size` in the body is a synonym for `scroll_size` as a URL parameter. This comes from how search works. An update by query has a body that is identical to a search, except for a few added options like the script. And in a scroll search, `size` means the scroll size (whereas in a regular search it just means the number of hits to get).

I think we should rename it to `scroll_size` or `batch_size` (with deprecation warnings in 7.x) for these APIs. Will probably cause less confusion than the fact that it is not identical to how it is specified in a scroll search. Also, with point-in-time arriving, scroll search should really no longer be used by clients.
